src/box/call.c

```diff
@@ -157,7 +158,7 @@ box_process_call(struct call_request *request, struct port *port)
 	}
 	if (rc != 0)
 		return -1;
-	if (in_txn() != NULL) {
+	if (in_txn() != NULL && stream_id == 0) {
 		diag_set(ClientError, ER_FUNCTION_TX_ACTIVE);
 		port_destroy(port);
 		return -1;
```

Comment:
Shouldn't we rollback the transaction here? The same question is for box_process_eval.

Reply:
For non-streams, we leave the old behavior for call and eval. For streams, we allow transactions via call and eval + sql requests (execute('TRANSACTION BEGIN')). Initially, we discussed it when we met with Mons.

Reply:
I was afraid the transaction might get detached in case stream_id == 0, but looking at tx_set_txn_after, I see that it won't be detached unless there's a stream so everything should be fine.